<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1" />

  <title>Hierarchical Conditioning of Diffusion Models Using Tree-of-Life for Studying Species Evolution</title>
  <link rel="stylesheet" href="css/main.css" />
  <link rel="icon" href="images/icons/favicon.ico" />
  <link rel="preconnect" href="https://fonts.googleapis.com">
  <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
  <link href="https://fonts.googleapis.com/css2?family=Noto+Sans+Display:ital,wght@0,400;0,700;1,400;1,700&display=swap" rel="stylesheet">
</head>

<body>
  <main>
    <div style="text-align: center">
      <details>
        <summary>More Research</summary>
        <div class="options">
          <p><a href="https://imageomics.github.io/bioclip/">BioCLIP</a></p>
          <p><a href="https://github.com/Imageomics/INTR">INTR</a></p>
        </div>
      </details>
    </div>
    <h1>Hierarchical Conditioning of Diffusion Models Using Tree-of-Life for Studying Species Evolution</h1>
    <h2 style="text-align: center;">ECCV 2024</h2>
    <p class="centered">
      <sup>1</sup><a href="https://mridulk97.github.io/">Mridul Khurana</a>,
      <sup>2</sup><a href="https://scholar.google.com/citations?user=pz2Nm8AAAAAJ&hl=en">Arka Daw</a>,
      <sup>1</sup><a href="https://people.cs.vt.edu/marufm/">M. Maruf</a>,
      <sup>1</sup><a href="https://www.uyedalab.com/">Josef Uyeda</a>,
      <sup>3</sup><a href="https://www.faculty.uci.edu/profile/?facultyId=6769">Wasila Dahdul</a>,
      <sup>1</sup><a href="https://scholar.google.com/citations?user=LyaJ880AAAAJ&hl=en">Caleb Charpentier</a>,
      <sup>4</sup><a href="https://scholar.google.com/citations?user=bRO-9m8AAAAJ&hl=en">Yasin Bakış</a>,
      <sup>4</sup><a href="https://scholar.google.com/citations?user=K4typXsAAAAJ&hl=en">Henry L. Bart Jr.</a>,
      <sup>5</sup><a href="https://scholar.google.com/citations?user=ZKnlvcoAAAAJ&hl=en">Paula Mabee</a>,
      <sup>6</sup><a href="https://scholars.duke.edu/person/Hilmar.Lapp">Hilmar Lapp</a>,
      <sup>7</sup><a href="https://renci.org/staff/james-balhoff/">James Balhoff</a>,
      <sup>8</sup><a href="https://sites.google.com/view/wei-lun-harry-chao">Wei-Lun (Harry) Chao</a>,
      <sup>9</sup><a href="https://www.cs.rpi.edu/~stewart/">Charles Stewart</a>,
      <sup>8</sup><a href="https://cse.osu.edu/people/berger-wolf.1">Tanya Berger-Wolf</a>,
      <sup>1</sup><a href="https://people.cs.vt.edu/karpatne/">Anuj Karpatne</a>
    </p>
    <p class="centered">
      <sup>1</sup>Virginia Tech,
      <sup>2</sup>Oak Ridge National Lab,
      <sup>3</sup>University of California, Irvine,
      <sup>4</sup>Tulane University,
      <sup>5</sup>Battelle,
      <sup>6</sup>Duke University,
      <sup>7</sup>University of North Carolina at Chapel Hill,
      <sup>8</sup>The Ohio State University,
      <sup>9</sup>Rensselaer Polytechnic Institute
    </p>

    <p class="text-sm centered">
      <a href="mailto:mridul@vt.edu">mridul@vt.edu</a>, 
      <a href="mailto:karpatne@vt.edu">karpatne@vt.edu</a>
    </p>

<!--     <p class="centered"><strong>ECCV 2024.</strong></p> -->

    <p class="centered">
      <!-- <a class="pill-button" href="https://huggingface.co/datasets/imageomics/TreeOfLife-10M">
        <img src="images/icons/huggingface.svg" /> Data
      </a> -->
      <a class="pill-button" href="https://arxiv.org/abs/2408.00160v1">
        <img src="images/icons/arxiv.svg" /> Paper
      </a>
      <a class="pill-button" href="https://huggingface.co/spaces/mridulk/hier-embed">
        <img src="images/icons/huggingface.svg" /> Demo
      </a>
      <a class="pill-button" href="https://github.com/Imageomics/phylo-diffusion">
        <img src="images/icons/github.svg" /> Code
      </a>

    </p>
    <figure>
      <img srcset="" src="images/phylo_diffusion_architecture.jpg" alt="" loading="lazy">
      <figcaption>
        Figure 1: Overview of Phylo-Diffusion framework. Every species in the tree of life (phylogenetic tree) is encoded to a 
        HIERarchical Embedding (HIER-Embed) comprising of four vectors (one for each phylogenetic level), which is used to 
        condition a latent diffusion model to generate synthetic images of the species. By structuring the embedding space with 
        phylogenetic knowledge, Phylo-Diffusion enables visualization of changes in the evolutionary traits of a species 
        (circled pink) upon perturbing its embedding.
      </figcaption>
    </figure>

    <h2 class="banded">Phylo-Diffusion</h2>

    <p>
      We introduce Graph as a modality to conditional Diffusion Models, Phylo-Diffusion, for understanding in evolutionary traits in biological specimens.
      We leverage this new modality and introduce two experiments for understanding evolutionary traits inspired by gene-knockout experiments:
      <ol>
        <li><b>Trait Masking:</b>  where we mask out one or more level embedding by substituting it with gaussian noise to verify that the embeddings learned capture hierarchical information.</li>
        <li><b>Trait Swapping:</b> which involves substitution of level-l embedding of a source species with the level-l embedding of a sibling subtree at an equivalent level to visualize the trait differences in the generated images before and after trait swapping that can help us understand the evolutionary traits that branched at a certain level.</li>
      </ol>
    </p>

    <figure style="text-align: center;">
      <div style="display: flex; justify-content: space-around; align-items: flex-start;">
        <figure style="margin: 0 10px;">
          <img src="images/trait_masking.jpg" alt="" loading="lazy" style="height: 300px; width: auto;">
          <figcaption>
            (a) Trait Masking
          </figcaption>
        </figure>
        
        <figure style="margin: 0 10px;">
          <img src="images/trait_swapping.jpg" alt="" loading="lazy" style="height: 300px; width: auto;">
          <figcaption>
            (b) Trait Swapping
          </figcaption>
        </figure>
      </div>
      
      <figcaption style="margin-top: 10px;">
        Figure 2: Schematic diagrams of the two proposed experiments for discovering evolutionary traits using Phylo-Diffusion.
      </figcaption>
    </figure>
    
    
        
    <h3>Demo</h3>
    <p>Please note that the demo is currently running on free CPU resources provided by Hugging Face, so it may take up to 10 minutes to generate an image. We're working on securing additional resources to speed up the process. Thank you for your patience! </p>
    <script type="module" src="https://gradio.s3-us-west-2.amazonaws.com/4.36.1/gradio.js"></script>
    <gradio-app src="https://mridulk-hier-embed.hf.space/"></gradio-app>

    <h2 class="banded" id="evaluation">Experiments</h2>
    <p>
      We evaluate Phylo-Diffusion and show that it achieves high-fidelity at par with the text-to-image and class-conditional diffusion models.
    </p>

    <h4>Phylo-Diffusion performs at par with state-of-the-art generative models</h4>
    <p><i>Scroll to see all results.</i></p>

    <table>
      <thead>
        <tr>
          <th>Model Type</th>
          <th>Method</th>
          <th>FID &darr;</th>
          <th>IS &uarr;</th>
          <th>Prec. &uarr;</th>
          <th>Recall &uarr;</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>GAN</td>
          <td>Phylo-NN</td>
          <td>28.08</td>
          <td>2.35</td>
          <td>0.625</td>
          <td>0.084</td>
        </tr>
        <tr>
          <td>Diffusion</td>
          <td>Class Conditional</td>
          <td>11.46</td>
          <td>2.47</td>
          <td>0.679</td>
          <td>0.359</td>
        </tr>
        <tr>
          <td>Diffusion</td>
          <td>Scientific Name</td>
          <td>11.76</td>
          <td>2.43</td>
          <td>0.683</td>
          <td>0.332</td>
        </tr>
        <tr>
          <td>Diffusion</td>
          <td>Phylo-Diffusion (ours)</td>
          <td>11.38</td>
          <td>2.53</td>
          <td>0.654</td>
          <td>0.367</td>
        </tr>
      </tbody>
      </table>

    <h3>Trait Masking</h3>

    <figure>

      <img srcset="" src="images/level3-lepomis.jpg" alt="" loading="lazy">
      <figcaption>
        Figure 3: Probability activations for <i>Lepomis</i> species at level 3 for Trait Masking. 
        The species highlighted in green represent the species belonging to that subtree at the given level
      </figcaption>
    </figure>
    
    <h3>Trait Swapping</h3>

    Figure 4 illustrates trait swapping for the source species <i>Noturus exilis</i> (left), where the
    information at Level-2 is swapped with that of a sibling subtree at Node B
    (right). The image in the center is generated using the trait swapped embedding.
    This visualization of the perturbed species helps us study the trait changes that
    would have branched out at level-2 between Node A and Node B.In the generated
    image (center), we observe the absence of barbels(whiskers), and the caudal fin
    (tail) is getting forked (or split) highlighted in pink, which are traits adopted
    from species in the subtree at B (<i>Notropis</i>). Whereas other fins like the dorsal,
    pelvic, and anal fin still resemble the source species <i>Noturus exilis</i> highlighted
    in green. The same is also reflected in the change of probability distribution
    after perturbations; the probability distribution of source species <i>Noturus exilis</i>
    decreases and the probability of it being a <i>Notropis</i> increases slightly.

    <figure>
      <img srcset="" src="images/trait_swapping_example.jpg" alt="" loading="lazy">
      <figcaption>
        Figure 4: Visualization of changes in traits after swapping information at Level 2 (Node A) 
        for <i>Noturus exilis</i> (left) with its sibling subtree at Node B(right) to generate perturbed species (center). 
        Traits shared with the source species are outlined in green, whereas those shared with the sibling subtree 
        at Node B are outlined in pink.
      </figcaption>
    </figure>
    


    <h2 id="comparision" class="banded">Comparision to PhyloNN</h2>


    <figure>
      <img srcset="" src="images/phylo_nn_compare_2.jpg" alt="" loading="lazy">
      <figcaption>
        Figure 5: Comparing Phylo-NN with Phylo-Diffusion for trait swapping
      </figcaption>
    </figure>

    <!-- <p>
      Why does BioCLIP work so well?
      We conduct an intrinsic evaluation to understand the representations learned by BioCLIP.
      We visualize BioCLIP and CLIP's representations for the 100K unseen images in the iNat21 validation set, using T-SNE to plot them in two-dimensions, coloring the points based on their class.
      In the figure below, (B) means <b>B</b>ioCLIP and (O) means <b>O</b>penAI's CLIP.
    </p>
    <p>
      At higher ranks like phylum, both CLIP and BioCLIP have good separation, but you can see that BioCLIP's representation is more fine-grained and contains a richer clustering structure.
      At lower ranks, BioCLIP produces far more separable features, while CLIP's features tend to be cluttered and lack a clear structure.
      This shows that BioCLIP has learned a rich feature representation following the hierarchical structure of the taxonomy, which helps explain its strong generalization across the tree of life.
    </p>
    <figure>
      <a href="files/intrinsic.pdf">
        <img srcset="" src="images/intrinsic_small.png" alt="" loading="lazy">
      </a>
      <figcaption>
        We show that <b>(B)</b>ioCLIP's representations are more fine-grained and contain a richer clustering structure than <b>(O)</b>penAI's CLIP.
        Click on the image to see the full resolution PDF, or check out our <a href="https://arxiv.org/abs/2311.18803">paper</a> for more details.
      </figcaption>
    </figure> -->

    <!-- <h2 class="banded" id="dataset">Dataset</h2>
    <p>
      TreeOfLife-10M is the largest and most diverse available dataset of biology images.
      We combine images from three sources, iNaturalist, BIOSCAN-1M, and Encyclopedia of Life (<a href="https://eol.org">EOL</a>, accessed 29 July 2023), to create a dataset of
      10M images, spanning 450K+ species.
      We train BioCLIP on TreeOfLife-10M and release the weights for public use.
    </p> -->



    <h2 class="banded">References</h2>
    <p>Please cite our paper if you use our code, data, model or results.</p>
    <pre class="reference">
      @article{khurana2024hierarchical,
        title={Hierarchical Conditioning of Diffusion Models Using Tree-of-Life for Studying Species Evolution},
        author={Khurana, Mridul and Daw, Arka and Maruf, M and Uyeda, Josef C and Dahdul, Wasila and Charpentier, Caleb and Bak{\i}{\c{s}}, Yasin and Bart Jr, Henry L and Mabee, Paula M and Lapp, Hilmar and others},
        journal={arXiv preprint arXiv:2408.00160},
        year={2024}
      }
    </pre>

    <p>Also consider citing LDM:</p>
    <pre class="reference">@inproceedings{rombach2022high,
      title={High-resolution image synthesis with latent diffusion models},
      author={Rombach, Robin and Blattmann, Andreas and Lorenz, Dominik and Esser, Patrick and Ommer, Bj{\"o}rn},
      booktitle={Proceedings of the IEEE/CVF conference on computer vision and pattern recognition},
      pages={10684--10695},
      year={2022}
    }</pre>


    <h2 class="banded">Acknowledgements</h2>
    <p>
      This work was supported by the <a href="https://imageomics.org">Imageomics Institute</a>, which is funded by the US National Science Foundation's Harnessing the Data Revolution (HDR) program under <a href="https://www.nsf.gov/awardsearch/showAward?AWD_ID=2118240">Award #2118240</a> (Imageomics: A New Frontier of Biological Information Powered by Knowledge-Guided Machine Learning). Any opinions, findings and conclusions or recommendations expressed in this material are those of the author(s) and do not necessarily reflect the views of the National Science Foundation.
    </p>
  </main>
  <script>
    const details = document.querySelector("details");
    document.addEventListener("click", function (e) {
      if (!details.contains(e.target)) {
        details.removeAttribute("open");
      }
    });
  </script>
</body>

</html>
